src/plugins/workspace/public/components/workspace_menu/workspace_menu.tsx

```diff
@@ -67,111 +82,92 @@ export const WorkspaceMenu = ({ coreStart }: Props) => {
-      workspaceToItem
+  const getWorkspaceListItems = (panelsWorkspaceList: WorkspaceObject[], itemType: string) => {
+    const workspaceListItems = panelsWorkspaceList.map((workspace) =>
+      workspaceToItem(workspace, itemType)
     );
-    workspaceListItems.push({
-      icon: <EuiIcon type="plus" />,
-      name: i18n.translate('core.ui.primaryNav.workspaceContextMenu.createWorkspace', {
-        defaultMessage: 'Create workspace',
-      }),
-      key: WORKSPACE_CREATE_APP_ID,
-      onClick: () => {
-        window.location.assign(
-          cleanWorkspaceId(
-            coreStart.application.getUrlForApp(WORKSPACE_CREATE_APP_ID, {
-              absolute: false,
-            })
-          )
-        );
-      },
-    });
-    workspaceListItems.push({
-      icon: <EuiIcon type="folderClosed" />,
-      name: i18n.translate('core.ui.primaryNav.workspaceContextMenu.allWorkspace', {
-        defaultMessage: 'All workspaces',
-      }),
-      key: WORKSPACE_LIST_APP_ID,
-      onClick: () => {
-        window.location.assign(
-          cleanWorkspaceId(
-            coreStart.application.getUrlForApp(WORKSPACE_LIST_APP_ID, {
-              absolute: false,
-            })
-          )
-        );
-      },
-    });
     return workspaceListItems;
   };
 
-  const currentWorkspaceButton = (
-    <>
-      <EuiListGroup style={{ width: 318 }} maxWidth={false}>
-        <EuiListGroupItem
-          iconType="spacesApp"
-          label={currentWorkspaceName}
-          onClick={openPopover}
-          extraAction={{
-            color: 'subdued',
-            onClick: openPopover,
-            iconType: isPopoverOpen ? 'arrowDown' : 'arrowRight',
-            iconSize: 's',
-            'aria-label': 'Show workspace dropdown selector',
-            alwaysShow: true,
-          }}
-        />
-      </EuiListGroup>
-    </>
+  const currentWorkspaceButton = currentWorkspace ? (
+    <EuiButtonEmpty flush="left" onClick={openPopover} data-test-subj="current-workspace-button">
+      <EuiAvatar
+        size="s"
+        type="space"
+        name={currentWorkspace.name}
+        color={currentWorkspace.color}
+        initialsLength={2}
+      />
+    </EuiButtonEmpty>
+  ) : (
+    <EuiButtonIcon
+      iconType="spacesApp"
+      onClick={openPopover}
+      aria-label="workspace-select-button"
+      data-test-subj="workspace-select-button"
+    />
   );
 
-  const currentWorkspaceTitle = (
-    <EuiFlexGroup alignItems="center">
-      <EuiFlexItem grow={true}>
-        <EuiText size="s">{currentWorkspaceName}</EuiText>
-      </EuiFlexItem>
-      <EuiFlexItem grow={false}>
-        <EuiButtonIcon
-          iconType="cross"
-          onClick={closePopover}
-          aria-label="close workspace dropdown"
-        />
-      </EuiFlexItem>
-    </EuiFlexGroup>
-  );
+  const allWorkspacesPanels = [
+    {
+      id: 0,
+      title: (
+        <span className="custom-title">
```

Comment:
I think you can use OuiText component, then don't need to do customize css https://oui.opensearch.org/1.7/#/display/text

Reply:
ContextMenu defaults to converting the title to uppercase, which can only be prevented through CSS.
<img width="771" alt="image" src="https://github.com/user-attachments/assets/c9578133-340c-44b4-bee0-68d38aac60ee">

Reply:
I did a bit investigation, I think we should not use ContextMenu here, but instead we need to use ListGroup component https://oui.opensearch.org/1.7/#/display/list-group

Reply:
Use ListGroup to the two lists, then maybe you can use OuiTitle component for the title text

Reply:
Has replaced context with listGroup.